File: CX/ClientApp/src/servicios/ServicioTipoContactoEncuesta.js
import { ProcesarDatosApi } from "./ApiFetch";
const controlador = "tipocontactoencuesta";

export const AgregarTipoContactoEncuesta = async (data) => {
    const url = `${controlador}/insertartipocontactoencuesta`;
    return await ProcesarDatosApi('POST', url, data);
}

export const ActualizarTipoContactoEncuesta = async (data) => {
    const url = `${controlador}/actualizartipocontactoencuesta`;
    return await ProcesarDatosApi('PUT', url, data);
}

export const InactivarTipoContactoEncuesta = async (id) => {
    const url = `${controlador}/eliminartipocontactoencuesta?idTipoContactoEncuesta=${id}`;
    return await ProcesarDatosApi('DELETE', url);
}

export const ObtenerTipoContactoEncuestaPorId = async (id) => {
    const url = `${controlador}/obtenertipocontactoencuestaporid/${id}`;
    return await ProcesarDatosApi('GET', url);
}

export const ObtenerTipoContactoEncuesta = async () => {
    const url = `${controlador}/obtenertipocontactoencuesta`;
    return await ProcesarDatosApi('GET', url);
}
export const ObtenerTipoContactoEncuestaActivos = async () => {
    const url = `${controlador}/obtenertipocontactoencuestaactivos`;
    return await ProcesarDatosApi('GET', url);
}

File: BussinesLogic/Servicios/ServicioTipoContactoEncuesta.cs
using DataAccess.StoredProcedures;
using Dominio.Dto;
using Dominio.Entiti;
using Dominio.Interfaces.Aplicacion.TipoContactoEncuesta;
using Dominio.Interfaces.Infraestructura.BaseDatos;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;

namespace BussinesLogic.Servicios
{
    public class ServicioTipoContactoEncuesta: IServicioTipoContactoEncuesta
    {
        public readonly IRepositorioTipoContactoEncuesta repositorioTipoContactoEncuesta;

        public ServicioTipoContactoEncuesta(IRepositorioTipoContactoEncuesta repositorioTipoContactoEncuesta)
        {
            this.repositorioTipoContactoEncuesta = repositorioTipoContactoEncuesta;
        }
        public async Task<DtoRespuestaSP> InsertarTipoContactoEncuesta(EntitiTipoContactoEncuesta entitiTipoContactoEncuesta)
        {
            return await this.repositorioTipoContactoEncuesta.InsertarTipoContactoEncuesta(entitiTipoContactoEncuesta);
        }
        public async Task<DtoRespuestaSP> ActualizarTipoContactoEncuesta(EntitiTipoContactoEncuesta entitiTipoContactoEncuesta)
        {
            return await this.repositorioTipoContactoEncuesta.ActualizarTipoContactoEncuesta(entitiTipoContactoEncuesta);
        }
        public async Task<DtoRespuestaSP> EliminarTipoContactoEncuesta(int idTipoContactoEncuesta)
        {
            return await this.repositorioTipoContactoEncuesta.EliminarTipoContactoEncuesta(idTipoContactoEncuesta);
        }
        public async Task<DtoTipoContactoEncuesta> ObtenerTipoContactoEncuestaPorID(int idTipoContactoEncuesta)
        {
            return await this.repositorioTipoContactoEncuesta.ObtenerTipoContactoEncuestaPorID(idTipoContactoEncuesta);
        }

        public async Task<List<DtoTipoContactoEncuesta>> ObtenerTipoContactoEncuesta()
        {
            return await this.repositorioTipoContactoEncuesta.ObtenerTipoContactoEncuesta();
        }

        public async Task<List<DtoTipoContactoEncuesta>> ObtenerTipoContactoEncuestaActivos()
        {
            return await this.repositorioTipoContactoEncuesta.ObtenerTipoContactoEncuestaActivos();
        }
    }
}
